### PR TITLE
fix: Check before writing if writing without a part file

### PR DESCRIPTION
**Pull request by juliushaertl**
When uploading to object storage and not using a part file the logic from https://github.com/nextcloud/files_accesscontrol/pull/170 is behaving badly these days as the unlink will put the file into the trash bin. We can actually check before writing the file to stream as mime types are never detected by content on object storage.

Steps to reproduce:
- Setup primary object storage
- Upload a large file (through chunked upload) that has its mime type blocked

Before:
See that there is a copy of the file still stored in the trash bin

After:
See that the file is not moved on chunk assembly at all

### CODE REVIEW

**Comment by juliushaertl** [comment]
🏓 @nickvergessen Covered by the integration tests now and ready for review

File: tests/Integration/features/bootstrap/WebDav.php [hunk 1]
```diff
@@ -946,6 +946,25 @@ public function userDeletesEverythingInFolder($user, $folder) {
...
+		$client = $this->getSabreClient($user);
+		$properties = [
+			'{DAV:}getetag'
+		];
+
+		$response = $client->propfind($this->makeSabrePath($user, 'trash', 'trashbin'), $properties, 1);
+
+		unset($response['/remote.php/dav/trashbin/' . $user . '/trash/']);
+
+		var_dump($response);
```
Comment (nickvergessen) [inline tests/Integration/features/bootstrap/WebDav.php:963]: *cough*

File: tests/Integration/features/bootstrap/WebDav.php [hunk 2]
```diff
@@ -946,6 +946,25 @@ public function userDeletesEverythingInFolder($user, $folder) {
...
+	 */
+	public function userSeesNoFilesInTheTrashbin($user) {
+		$client = $this->getSabreClient($user);
+		$properties = [
+			'{DAV:}getetag'
+		];
+
+		$response = $client->propfind($this->makeSabrePath($user, 'trash', 'trashbin'), $properties, 1);
+
+		unset($response['/remote.php/dav/trashbin/' . $user . '/trash/']);
```
Comment (nickvergessen) [inline tests/Integration/features/bootstrap/WebDav.php:961]: This is the root entry?
Can we leave a comment for that?

Comment (juliushaertl) [reply tests/Integration/features/bootstrap/WebDav.php:961]: Done

**Comment by juliushaertl** [comment]
Preparing backports of integration tests so we can also have the stable branches covered.

**Pull request merged**